### PR TITLE
Adding 'model' property to ChatCompletions.

### PR DESCRIPTION
**Pull request by richardpark-msft**
`ChatCompletions` (the response from GetChatCompletions) is missing the `model` property that's in OpenAI's docs and in our example output in our docs. 

This adds it back in, with the C# project to rename it to DeploymentName.

- Our docs: https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#example-response-2
- OpenAI docs: https://platform.openai.com/docs/api-reference/chat/object#chat/object-model

### CODE REVIEW

File: specification/cognitiveservices/OpenAI.Inference/models/completions/chat_completions.tsp [hunk 1]
```diff
@@ -303,6 +304,12 @@ model ChatCompletions {
     """)
   choices: ChatChoice[];
 
+  @doc("""
+    The deployment name used for this completions request.
+  """)
+  @clientName("DeploymentName", "csharp")
```
Comment (johanste) [inline specification/cognitiveservices/OpenAI.Inference/models/completions/chat_completions.tsp:310]: I do not believe that this is the deployment name in AOAI.

Comment (richardpark-msft) [reply specification/cognitiveservices/OpenAI.Inference/models/completions/chat_completions.tsp]: Fixed - removed the clientName and also updated the doc comment to say it's a model.

**Review by johanste — APPROVED** [review]

**Review by kristapratico — APPROVED** [review]

**Comment by mario-guerra** [comment]
@richardpark-msft is the service team aware of and in agreement with these changes?

**Comment by richardpark-msft** [comment]
> @richardpark-msft is the service team aware of and in agreement with these changes?

I'd worked with @trrwilson and @johanste on this. Apparently this was an SDK omission and, now that we have a good use case for it, we're just bringing it back.

**Pull request merged**